app/commands/cart_services/resolve_cart_item.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+#
+# Copyright 2021 Victoria Garcia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class CartServices::ResolveCartItem
+  include ThemeConcern
+
+  MEMBERSHIP = CartItem::MEMBERSHIP
+  UNKNOWN = CartItem::UNKNOWN
+
+  def initialize(full_params: f_params, bin_for_now: n_cart, item_kind: i_kind)
+    @full_params = full_params
+    @now_bin = bin_for_now
+    @item_kind = item_kind
+    @benefitable_required = benefitable_required?(item_kind)
+    @error = ""
+    @offer_param = full_params[:offer]
+  end
+
+  def call
+    our_acquirable = resolve_acquirable
+    return {cart_item: nil, error: @error} if our_acquirable.blank?
+
+    our_benefitable = resolve_benefitable
+    return {cart_item: nil, error: @error} if ((our_benefitable.blank? && @benefitable_required) || !our_benefitable.valid?)
+
+    our_item = create_cart_item(our_acquirable, benefitable: our_benefitable)
+    return {cart_item: our_item, error: @error}
+  end
+
+  private
+
+  def benefitable_required?(item_kind)
```

Comment:
What does "benefitable" mean?

(also, a general point in these: since the alternate branches of these conditions are invalid states right now (we don't have other items) we should make them fail, rather than silently do nothing.

Reply:
Benefitable is a polymorphic association type that, in this case, represents the beneficiary of a membership.  I figure that Benefitable will also be used in the case of something like a t-shirt sale, where the recipient and the user aren't necessarily the same person.  (Or, say, if someone buys site selection tokens for someone else?  Though I don't even know if that's a sensible concept.)